### PR TITLE
Add doc for updating target platform in .ci.yaml

### PR DESCRIPTION
**Pull request by keyonghan**
Related https://github.com/flutter/flutter/issues/125492

### CODE REVIEW

File: CI_YAML.md [hunk 1]
```diff
@@ -180,6 +180,38 @@ tags: >
...
+2. Add a new entry under the new platform as `bringup: true` with necessary dependencies
+3. Land the change with the new entry
+4. If the new target under the new platform passes in postsubmit
+  1. Remove the old target entry and mark the new target as `bringup: false`
+
+Example: say one wants to switch `Linux_android web_size__compile_test` to a vm.
+
+Existing config:
+```yaml
+- name: Linux_android web_size__compile_test
```
Comment (zanderso) [inline CI_YAML.md:196]: Could you explain the part of this change that specifically causes the new version to become host-only? Is it removing the `_android` from the name, or is it some other part of the config?

(As an aside, encoding the hardware requirements in the config name seems confusing to me, if that's the case. Should it be explicitly specified in the `properties` instead?)

Comment (keyonghan) [reply CI_YAML.md:196]: Yes, by updating the prefix platform in the target name. See the target `name` described above: https://github.com/flutter/cocoon/blob/main/CI_YAML.md#ciyaml

This dependency of platform in the target name is some legacy thing. https://github.com/flutter/flutter/issues/114905 tracks deprecating the platform dependency on target names.

Comment (zanderso) [reply CI_YAML.md:196]: I'd suggest including that explanation in the docs in this PR. Under `Linux_android web_size__compile_test` I'm currently also seeing:
```
properties:
      tags: >
        ["devicelab", "android", "linux"]
```
Do any of those need to stay or be removed in order to make the test host-only?

Comment (keyonghan) [reply CI_YAML.md:202]: Updated the doc accordingly, including the `tags`.

File: CI_YAML.md [hunk 2]
```diff
@@ -180,6 +180,49 @@ tags: >
...
+
+Target depends on the prefix platform in its `name` to decide which platform to run on. This should match
+to an existing platform under `platform_properties`.
+
+If one target needs to switch running platforms, e.g. from a devicelab bot to a host only bot:
+1. Keep the old target entry
+2. Add a new entry under the new platform with
+  1. `bringup: true`
+  2. necessary dependencies
+  3. corresponding tags (tags will only be used for infra metrics analysis)
```
Comment (ricardoamador) [inline CI_YAML.md:193]: Aren't tags also used for performance tests or is that what you mean here?

**Review by zanderso — APPROVED** [review]
Thanks!

**Pull request merged**